+nosnoc/+discrete_time_problem/Gcs.m

```diff
@@ -0,0 +1,870 @@
+                        ubh = (1+opts.gamma_h)*h0*opts.s_sot_max;
+                        lbh = (1-opts.gamma_h)*h0/opts.s_sot_min;
+                    end
+                    obj.w.h(ii,1:opts.N_finite_elements(ii)) = {{'h', 1}, lbh, ubh, h0};
+                end
+                if obj.opts.step_equilibration == StepEquilibrationMode.linear_complementarity
+                    obj.w.B_max(ii,2:opts.N_finite_elements(ii)) = {{'B_max', dims.n_lambda},-inf,inf};
+                    obj.w.pi_c(ii,2:opts.N_finite_elements(ii)) = {{'pi_c', dims.n_c},-inf,inf};
+                    obj.w.pi_lambda(ii,2:opts.N_finite_elements(ii)) = {{'pi_lambda', dims.n_lambda},-inf,inf};
+                    obj.w.lambda_c(ii,2:opts.N_finite_elements(ii)) = {{'lambda_c', dims.n_c},0,inf};
+                    obj.w.lambda_lambda(ii,2:opts.N_finite_elements(ii)) = {{'lambda_lambda', dims.n_lambda},0,inf};
+                    obj.w.eta(ii,2:opts.N_finite_elements(ii)) = {{'eta', dims.n_lambda},0,inf};
+                    obj.w.nu(ii,2:opts.N_finite_elements(ii)) = {{'nu', 1},0,inf};
+                end
+                if ~opts.right_boundary_point_explicit
+                    obj.w.not_c_primals(ii,1:opts.N_finite_elements(ii)) = {{'not_c_primal', 2}, 0 inf, 0.5};
+                    obj.w.mu_not_c(ii,1:opts.N_finite_elements(ii)) = {{'mu_lambda', 3}, 0 inf, 0.5};
+                end
+            end
+
+            % For c_n ~= 1 case
+            rbp = ~opts.right_boundary_point_explicit;
+            
+            % 3d vars: Variables defined on each rk stage
+            %          some of which are also defined at the initial point:
+            obj.w.x(0,0,opts.n_s) = {{['x_0'], dims.n_x}, model.x0, model.x0, model.x0};
+            obj.w.z(0,0,opts.n_s) = {{'z', dims.n_z}, model.lbz, model.ubz, model.z0};
+            obj.w.lambda(0,0,opts.n_s) = {{['lambda'], dims.n_lambda},0,inf};
+            for ii=1:opts.N_stages
+                if (opts.rk_representation == RKRepresentation.integral ||...
+                    opts.rk_representation == RKRepresentation.differential_lift_x)
+                    obj.w.x(ii,1:opts.N_finite_elements(ii),1:(opts.n_s+rbp)) = {{'x', dims.n_x}, model.lbx, model.ubx, model.x0};
+                else
+                    obj.w.x(ii,1:opts.N_finite_elements(ii),opts.n_s) = {{'x', dims.n_x}, model.lbx, model.ubx, model.x0};
+                end
+                if (opts.rk_representation == RKRepresentation.differential ||...
+                    opts.rk_representation == RKRepresentation.differential_lift_x)
+                    obj.w.v(ii,1:opts.N_finite_elements(ii),1:opts.n_s) = {{'v', dims.n_x}};
+                end
+                
+                obj.w.z(ii,1:opts.N_finite_elements(ii),1:(opts.n_s+rbp)) = {{'z', dims.n_z}, model.lbz, model.ubz, model.z0};
```

Comment:
If the function c is nonlinear, then it makes sense to lift it (as we know comp. constraints dont like if not lifted)
how is this happening currently, at the mpcc level? or is somewhere a algebraic variable z_c (or whatever name) defined and an algebaric constraint to take care of it (i feel like i missed it in dcs.GCS, but is to me maybe acontinious time think for lifting  - for lifting theta and alpha, in Heaviside we also do it all before discretization)

Reply:
I have not implemented lifting as of now as I have not seen any major difficulties from even nonlinear $c(x)$. However, it is probably a good idea to provide as an option.

Reply:
No difficulties, but it may improve performance. I am curious to learn about the experiences. 
Our guiding principle in general is anyway: affine functions into complementarity constraints.